app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,62 @@
+class OrdersController < ApplicationController
+  before_action :find_order, except: [:new, :create]
+
+  def find_order
+    @order = Order.find(params[:id])
+  end
+
+  # connect products order item controller to instantiate an order item
+  # and then to the order controller to shovel into the order
+  def new
+    @order = Order.new
+    # triggered by the 'add to cart' button on Products#show
+  end
+
+  # def find_or_create
+  #   @order = Order.where(params[:id]).first_or_create do |order|
+  #     @order_item = OrderItem.create!(product_id: (Product.find(params[:id]).id)
+  #     order.order_items << @order_item
+  #     order.save
+  #   end
+  #   # redirect_to #order_cart_path
+  # end
+
+  def show
+    @order_items = @order.order_items
+  end
+
+  def payment
+    # check for requested amount of inventory & update before completing payment
+    update_inventory(@order)
+
+    @order.email = params[:order][:email]
+    @order.address1 = params[:order][:address1]
+    @order.address2 = params[:order][:address2]
+    @order.city = params[:order][:city]
+    @order.state = params[:order][:state]
+    @order.zipcode = params[:order][:zipcode]
+    @order.card_last_4 = params[:order][:card_number.split(//)][-1, 4].join(",")
```

Comment:
Two notes:
- this should probably be a model callback (`before_validation`, most likely) that ensures the data is in the right format.
- Strings can be indexed like arrays. I'm pretty sure `"1234567890"[-4,4] #=> "7890"`